### PR TITLE
Library: update to plugin core v0.11.0

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
+    chrome: 'readonly',
+  },
+
+  ignorePatterns: [
+    'docs/*',
+    'dist/*',
+    'src/assets/index.js',
+  ],
+
+  // add your custom rules here
```
Comment (k-y0u) [inline .eslintrc.js]: Add

File: .eslintrc.js [hunk 2]
```diff
@@ -1,16 +1,70 @@
...
+    'import/default': 'error',
+    'import/export': 'error',
+    'import/extensions': 'off',
+    'import/no-unresolved': 'off',
+    'import/no-extraneous-dependencies': 'off',
+    'import/prefer-default-export': 'off',
+
+    'prefer-promise-reject-errors': 'off',
+
+    // allow debugger during development only
```
Comment (k-y0u) [inline .eslintrc.js]: Allows

File: jsdoc.config.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+	"plugins": [],
```
Comment (k-y0u) [inline jsdoc.config.json]: indent = 2 spaces

File: scripts/generateResources.js [hunk 4]
```diff
@@ -1,23 +0,0 @@
-const fs = require('fs');
```
Comment (k-y0u) [inline scripts/generateResources.js:1]: update commit name... things is not accurate and professional enough.

File: src/draw/MyPluginDrawer.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { DefaultDrawer } from 'leto-modelizer-plugin-core';
+
```
Comment (k-y0u) [inline src/draw/MyPluginDrawer.js:2]: update commit message.

File: src/metadata/MyPluginMetadata.js [hunk 6]
```diff
@@ -14,16 +14,18 @@ class MyPluginMetadata extends DefaultMetadata {
...
 
   /*
-   * Implement this to provide all the definitions describing the component.
+   * Implement this to provide all the definitions describing the components.
    *
    * ComponentDefinition is used to generate the instantiable component list.
    *
    * And the componentAttributeDefinition is used to generate the form to update a component.
    *
    * Both of them can be also used to check components in parser and generate error.
```
Comment (k-y0u) [inline src/metadata/MyPluginMetadata.js]: errors

File: src/models/MyPlugin.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  DefaultData,
+} from 'leto-modelizer-plugin-core';
+import MyPluginDrawer from 'src/draw/MyPluginDrawer';
+import MyPluginMetadata from 'src/metadata/MyPluginMetadata';
+import MyPluginParser from 'src/parser/MyPluginParser';
+import MyPluginRenderer from 'src/render/MyPluginRenderer';
+import { name, version } from 'package.json';
+
+/**
+ * My plugin.
```
Comment (k-y0u) [inline src/models/MyPlugin.js]: Example of a potential plugin model.
OR
Template of plugin model.

File: src/parser/MyPluginParser.js [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+  isParsable(fileInformation) {
     /*
      * Implement this to indicate which fileName your provider can manage.
      *
      * You just have to return a Boolean to say if your parser can manage the file or not.
      *
      * By default, this function return false only on null/undefined fileName.
      */
-    return super.isParsable(fileName);
+    return super.isParsable(fileInformation);
```
Comment (k-y0u) [inline src/parser/MyPluginParser.js:15]: update description too.

File: src/render/MyPluginRenderer.js [hunk 9]
```diff
@@ -1,16 +1,18 @@
...
 
 class MyPluginRenderer extends DefaultRender {
   /*
    * The purpose of this function is to generate the content of one file.
    *
-   * You have to map all the given components and links into a file content.
+   * You have to map all the given components into a file content.
    */
-  render(components = [], links = []) {
+  renderFiles() {
```
Comment (k-y0u) [inline src/render/MyPluginRenderer.js:7]: update description.

File: src/assets/index.js [hunk 10]
```diff
@@ -1 +0,0 @@
-export default {};
```
Comment (k-y0u) [inline src/assets/index.js:1]: update commit message (resize icon too and add defaultContainer)

File: README.md [hunk 11]
```diff
@@ -125,26 +157,15 @@ You have to implement you own parser to manage the file formats you want.
...
-```json
-{
-  "components": [],
-  "links": [],
-  "errors": [],        
-}
-```
+- `parse(inputs[])`, to parse an array of file input and retrieve all the corresponding components.
+
+The parse method update the `this.pluginData.components` and `this.pluginData.parseErrors`.
```
Comment (k-y0u) [inline README.md]: updates

File: README.md [hunk 12]
```diff
@@ -125,26 +157,15 @@ You have to implement you own parser to manage the file formats you want.
...
-- `components` is an array of Component.
-- `links` is an array of ComponentLinks.
-- `errors` is an array of ParseError.
-- 
 You can use any library you want to parse all the files. For your information, we use [Antlr](https://www.antlr.org/) for our terraform parser.
 
 ### Manage the parser errors  
 
-On LetoModelizer we use Monaco editor and we design our ParseError class to make all errors usable by MonacoEditor.
+On LetoModelizer we use Monaco editor, we design our ParseError class to make all errors usable by MonacoEditor.
```
Comment (k-y0u) [inline README.md]: bad modification. `and` OR `. We`

File: README.md [hunk 13]
```diff
@@ -166,60 +187,28 @@ But for model, if you want the drawer to set the default values on it, you have
...
-```
+- a class `.component-type`: to set the component type as text
 
 ### Use custom model in drawer
 
 You can provide multiple models and icons that can be used by the drawer.
 
-To do that, you have to add the svg of your icon in the folder `src/assets/icons` and the svg of your
-model in the folder `src/assets/models`.
+To do that, you have to add the svg of your icon in the folder `public/icons` and the svg of your
```
Comment (k-y0u) [inline README.md]: your icons

File: README.md [hunk 14]
```diff
@@ -166,60 +187,28 @@ But for model, if you want the drawer to set the default values on it, you have
...
+model in the folder `public/models`.
```
Comment (k-y0u) [inline README.md]: models

File: README.md [hunk 15]
```diff
@@ -166,60 +187,28 @@ But for model, if you want the drawer to set the default values on it, you have
...
 
 Take care of the model name, it will be used by the specific method called in drawer.
 
 Example:
 
-Add model `MyFruitModel.svg` in `src/assets/models/` and an icon `banana.svg` in `src/assets/icons`.
+Add model `MyFruitModel.svg` in `public/models` and an icon `banana.svg` in `public/icons`.
```
Comment (k-y0u) [inline README.md]: a model

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (Zorin95670) [reply .eslintrc.js]: done

File: .eslintrc.js (hunk 2 above)
Comment (Zorin95670) [reply .eslintrc.js]: done

File: jsdoc.config.json (hunk 3 above)
Comment (Zorin95670) [reply jsdoc.config.json]: done

File: src/metadata/MyPluginMetadata.js (hunk 6 above)
Comment (Zorin95670) [reply src/metadata/MyPluginMetadata.js]: done

File: src/models/MyPlugin.js (hunk 7 above)
Comment (Zorin95670) [reply src/models/MyPlugin.js]: done

File: src/parser/MyPluginParser.js (hunk 8 above)
Comment (Zorin95670) [reply src/parser/MyPluginParser.js:15]: done

File: src/render/MyPluginRenderer.js (hunk 9 above)
Comment (Zorin95670) [reply src/render/MyPluginRenderer.js:7]: done

File: README.md (hunk 11 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 12 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 13 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 14 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 15 above)
Comment (Zorin95670) [reply README.md]: done

File: src/draw/MyPluginDrawer.js (hunk 5 above)
Comment (Zorin95670) [reply src/draw/MyPluginDrawer.js:2]: done

File: scripts/generateResources.js (hunk 4 above)
Comment (Zorin95670) [reply scripts/generateResources.js:1]: done

File: src/parser/MyPluginParser.js [hunk 16]
```diff
@@ -1,15 +1,18 @@
-import {DefaultParser} from 'leto-modelizer-plugin-core';
+import { DefaultParser } from 'leto-modelizer-plugin-core';
 
+/**
+ * Template of plugin parser.
+ */
 class MyPluginParser extends DefaultParser {
-  isParsable(fileName) {
+  isParsable(fileInformation) {
```
Comment (k-y0u) [inline src/parser/MyPluginParser.js:7]: function description missing

File: src/parser/MyPluginParser.js [hunk 17]
```diff
@@ -1,15 +1,18 @@
...
     /*
-     * Implement this to indicate which fileName your provider can manage.
+     * Implement this to indicate which fileInformation your provider can manage.
      *
      * You just have to return a Boolean to say if your parser can manage the file or not.
      *
      * By default, this function return false only on null/undefined fileName.
```
Comment (k-y0u) [inline src/parser/MyPluginParser.js]: update `filename` to `fileInformation name` or something else.

File: src/render/MyPluginRenderer.js [hunk 18]
```diff
@@ -1,16 +1,21 @@
...
+/**
+ * Template of plugin renderer.
+ */
 class MyPluginRenderer extends DefaultRender {
   /*
-   * The purpose of this function is to generate the content of one file.
+   * The purpose of this function is to generate all files.
    *
-   * You have to map all the given components and links into a file content.
+   * You have to map all the given components into a file content and return all files.
```
Comment (k-y0u) [inline src/render/MyPluginRenderer.js]: missing return description

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/MyPluginParser.js (hunk 17 above)
Comment (Zorin95670) [reply src/parser/MyPluginParser.js]: done

**Review by k-y0u — APPROVED** [review]

**Pull request merged**